packages/flutter/lib/src/material/menu_theme.dart

```diff
@@ -0,0 +1,354 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'material_state.dart';
+import 'menu_bar.dart';
+import 'theme.dart';
+
+// Examples can assume:
+// const Widget child = SizedBox();
+
+/// Defines the visual properties of [MenuBar], [MenuButton] and
+/// [MenuItemButton] widgets.
+///
+/// Descendant widgets obtain the current [MenuThemeData] object
+/// using `MenuTheme.of(context)`. Instances of
+/// [MenuThemeData] can be customized with
+/// [MenuThemeData.copyWith].
+///
+/// Typically, a [MenuThemeData] is specified as part of the
+/// overall [Theme] with [ThemeData.menuTheme]. Otherwise,
+/// [MenuTheme] can be used to configure its own widget subtree.
+///
+/// All [MenuThemeData] properties are `null` by default.
+/// If any of these properties are null, the menu bar will provide its own
+/// defaults.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme for the application.
+@immutable
+class MenuThemeData with Diagnosticable {
```

Comment:
Developers will need to override the appearance of cascading menus, popup menus, and dropdown menus, and perhaps separately menu bars. It might not make sense to call anything just "menuTheme".

Reply:
Okay.  Should I just make a plethora of themes, then?

Reply:
Themes for MenuBar and MenuButton sounds reasonable. We'll be adding a DropdownTheme (or maybe ComboBoxTheme if we find that we need a new name for the M3 version). There already is a PopupMenu theme.